app/controllers/discussions_controller.rb

```diff
@@ -48,14 +51,25 @@ def update
 
   private
 
+    def check_discussion_is_upcoming
+      if !(@discussion.date.to_s > Date.current.to_s)
```

Comment:
in any case `@discussion.date` would be a datetime object and so would be `Date.current`. I don't think you need to convert it to string.
Also, we can govern how what `to_s` does to date objects. Search it and lets discuss.